workers/price-reporter/src/manager.rs

```diff
@@ -1,20 +1,63 @@
 //! Defines the PriceReporterExecutor, the handler that is responsible
 //! for executing individual PriceReporterJobs.
-use common::default_wrapper::{DefaultOption, DefaultWrapper};
-use common::types::exchange::PriceReporterState;
-use common::types::token::Token;
-use common::types::CancelChannel;
-use common::{new_async_shared, AsyncShared};
-use job_types::price_reporter::{PriceReporterJob, PriceReporterReceiver};
-use std::{collections::HashMap, thread::JoinHandle};
-use tokio::runtime::Runtime;
-use tokio::sync::oneshot::Sender as TokioSender;
-use tracing::{error, info, info_span, warn, Instrument};
-use util::err_str;
-
-use crate::errors::{ExchangeConnectionError, PriceReporterError};
-
-use super::{reporter::Reporter, worker::PriceReporterConfig};
+
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    thread::JoinHandle,
+};
+
+use atomic_float::AtomicF64;
+use common::types::{
+    exchange::{Exchange, PriceReport, PriceReporterState},
+    token::Token,
+    Price,
+};
+use statrs::statistics::{Data, Median};
+use util::get_current_time_seconds;
+
+use crate::{errors::PriceReporterError, worker::PriceReporterConfig};
+
+// pub mod external_executor;
+pub mod native_executor;
+
+// -------------
+// | Constants |
+// -------------
+
+/// If Binance has not reported an update within
+/// MAX_REPORT_AGE (in milliseconds), we pause matches until we receive a more
+/// recent price. Note that this threshold cannot be too aggressive, as certain
+/// long-tail asset pairs legitimately do not update that often.
+const MAX_REPORT_AGE_MS: u64 = 20_000; // 20 seconds
+/// If we do not have at least MIN_CONNECTIONS reports, we pause matches until
+/// we have enough reports. This only applies to Named tokens, as Unnamed tokens
+/// simply use UniswapV3.
+const MIN_CONNECTIONS: usize = 1;
+/// If a Binance PriceReport is more than MAX_DEVIATION (as a fraction) away
+/// from the midpoint, then we pause matches until the prices stabilize.
+const MAX_DEVIATION: f64 = 0.02;
```

Comment:
2% is quite high, is this just a dummy value? we'd like to push this lower IMO

Reply:
This is a dummy, value, though Chris and I discussed using 1%, so I'll set that here